### PR TITLE
Convert WPP to NET Core 3.1

### PR DESCRIPTION
**Pull request by funjoker**
- Copyright Update
- Hotfix Parsing
- DB2Hash update
- Update DBC/DB2 Reader (under MIT license)

#292

### CODE REVIEW

**Comment by funjoker** [comment]
no idea about coverity

**Comment by Aokromes** [comment]
i removed coverity years ago because no one cared about it, no?

**Comment by funjoker** [comment]
no idea about `dotnet test` in travis

File: WowPacketParser/SQL/SQLDatabase.cs [hunk 1]
```diff
@@ -94,54 +94,74 @@ private static void LoadBroadcastText()
             string query =
                 "SELECT ID, Text, Text1, EmoteID1, EmoteID2, EmoteID3, EmoteDelay1, EmoteDelay2, EmoteDelay3, EmotesID, LanguageID, Flags, ConditionID, SoundEntriesID1, SoundEntriesID2 " +
                 $"FROM {Settings.HotfixesDatabase}.broadcast_text;";
-            using (var reader = SQLConnector.ExecuteQuery(query))
+
+            if (Settings.TargetedDatabase < TargetedDatabase.Cataclysm)
```
Comment (Shauren) [inline WowPacketParser/SQL/SQLDatabase.cs]: 3.3.5 branch also has broadcast_text table in world db

Comment (funjoker) [reply WowPacketParser/SQL/SQLDatabase.cs]: solved

**Comment by funjoker** [comment]
>   X TestOperators [47ms]
>   Error Message:
>      Expected: 1
>   But was:  True
>   Stack Trace:
>      at WowPacketParser.Tests.Misc.BitTest.TestOperators() in /home/travis/build/TrinityCore/WowPacketParser/WowPacketParser.Tests/Misc/BitTest.cs:line 23

Need advice here

**Comment by funjoker** [comment]
It's done!

**Comment by funjoker** [comment]
`_startTime` is TimeDataKind.Utc when read out of header.
But in optionalData it was converted to local time already, this leaded in `Packet Read` to convert it again to local time.

Keeping both as UTC and converting them later to local, we avoid issues.

**Comment by funjoker** [comment]
Done

**Comment by jackpoz** [comment]
macOS image on appveyor has .NET Core 3.1 SDK now https://github.com/appveyor/ci/issues/3311

**Comment by Aokromes** [comment]
WowPacketParser/WowPacketParser.csproj
WowPacketParserModule.V8_0_1_27101/WowPacketParserModule.V8_0_1_27101.csproj

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**